src/coreclr/jit/importer.cpp

```diff
@@ -15319,41 +15278,27 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                         usesHelper = true;
                         break;
 
+                    case CORINFO_FIELD_STATIC_SHARED_STATIC_HELPER:
                     case CORINFO_FIELD_STATIC_ADDRESS:
                         // Replace static read-only fields with constant if possible
                         if ((aflags & CORINFO_ACCESS_GET) && (fieldInfo.fieldFlags & CORINFO_FLG_FIELD_FINAL) &&
-                            !(fieldInfo.fieldFlags & CORINFO_FLG_FIELD_STATIC_IN_HEAP) &&
-                            (varTypeIsIntegral(lclTyp) || varTypeIsFloating(lclTyp)))
+                            !(fieldInfo.fieldFlags & CORINFO_FLG_FIELD_STATIC_IN_HEAP))
```

Comment:
```suggestion
```
This condition is unnecessary to check in the JIT. It is up to the EE side to decide whether it can figure out the value for this case.

Reply:
Right, I left it just in hopes it would slightly improve JIT TP, but I assume it's invisible and it will e.g. stop structs I plan to add eventually.